src/components/picker/PickerItem.js

```diff
@@ -16,153 +16,138 @@ import Text from '../text';
-
-    return (
-      <TouchableOpacity
-        activeOpacity={0.5}
-        onPress={this.onPress}
-        onLayout={isSelected ? this.onSelectedLayout : undefined}
-        disabled={disabled}
-        testID={testID}
-        throttleTime={0}
-        {...this.extractAccessibilityProps()}
-      >
-        {renderItem ? renderItem(value, this.props, this.getLabel()) : this.renderItem()}
-      </TouchableOpacity>
-    );
+  return (
+    <TouchableOpacity
+      activeOpacity={0.5}
+      onPress={_onPress}
+      onLayout={isSelected ? onSelectedLayout : undefined}
+      disabled={disabled}
+      testID={testID}
+      throttleTime={0}
+      {...Modifiers.extractAccessibilityProps(props)}
+    >
+      {renderItem ? renderItem(value, props, getLabel()) : _renderItem()}
+    </TouchableOpacity>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    height: 56.5,
+    paddingHorizontal: 23,
+    borderColor: Colors.rgba(Colors.dark10, 0.1),
+    borderBottomWidth: 1
+  },
+  labelText: {
+    ...Typography.text70,
+    color: Colors.dark10,
+    flex: 1,
+    textAlign: 'left'
+  },
+  labelTextDisabled: {
+    color: Colors.dark60
   }
-
-  // TODO: deprecate the check for object
-  onPress = () => {
-    const {value, onPress} = this.props;
-    // onPress(_.isObject(value) ? value : {value, label});
-    onPress(value);
-  };
-}
-
-function createStyles() {
-  return StyleSheet.create({
-    container: {
-      height: 56.5,
-      paddingHorizontal: 23,
-      borderColor: Colors.rgba(Colors.dark10, 0.1),
-      borderBottomWidth: 1
-    },
-    labelText: {
-      ...Typography.text70,
-      color: Colors.dark10,
-      flex: 1,
-      textAlign: 'left'
-    },
-    labelTextDisabled: {
-      color: Colors.dark60
-    }
-  });
-}
+});
+
+PickerItem.displayName = 'Picker.Item';
+PickerItem.propTypes = {
+  /**
+   * Item's value
+   */
+  value: PropTypes.oneOfType([PropTypes.object, PropTypes.string, PropTypes.number]),
+  /**
+   * Item's label
+   */
+  label: PropTypes.string,
+  /**
+   * Custom function for the item label (e.g (value) => customLabel)
+   */
+  getItemLabel: PropTypes.func,
+  /**
+   * DEPRECATE: Function to return the value out of the item value prop when value is custom shaped.
+   */
+  getItemValue: PropTypes.func,
+  /**
+   * Is the item selected
+   */
+  isSelected: PropTypes.bool,
+  /**
+   * Pass to change the selected icon
+   */
+  selectedIcon: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
+  /**
+   * Pass to change the selected icon color
+   */
+  selectedIconColor: PropTypes.string,
+  /**
+   * Is the item disabled
+   */
+  disabled: PropTypes.bool,
+  /**
+   * Render custom item
+   */
+  renderItem: PropTypes.elementType,
+  /**
+   * Callback for onPress action
+   */
+  onPress: PropTypes.func,
+  /**
+   * Callback for onLayout event
+   */
+  onSelectedLayout: PropTypes.func
+};
 
 export default PickerItem;
```

Comment:
Due tho some of the of the props in PickerItem (like `renderItem` callback prop), it's very possible that memoizing the whole component might cause some side effects and break behaviors. 
For now I think it will be better to keep parity with the previous implementation (where PickerItem was simple Component and not a pure one) and later on consider optimizations if necessary.

Reply:
Understood :)